bootstrap.sh

```diff
@@ -1,77 +1,159 @@
 
     clangFormatUrl=https://clrjit.blob.core.windows.net/clang-tools/${info}/clang-format
 
-    if validate_url ${clangFormatUrl} > /dev/null;
-    then
+    if validate_url ${clangFormatUrl} > /dev/null; then
         echo "Downloading clang-format to bin directory"
         # download appropriate version of clang-format
         wget ${clangFormatUrl} -O bin/clang-format
         chmod 751 bin/clang-format
+    else
+        echo "clang-format not found here: ${clangFormatUrl}"
     fi
 
     clangTidyUrl=https://clrjit.blob.core.windows.net/clang-tools/${info}/clang-tidy
 
-    if validate_url ${clangTidyUrl} > /dev/null;
-    then
+    if validate_url ${clangTidyUrl} > /dev/null; then
         echo "Downloading clang-tidy to bin directory"
         # download appropriate version of clang-tidy
         wget ${clangTidyUrl} -O bin/clang-tidy
         chmod 751 bin/clang-tidy
+    else
+        echo "clang-tidy not found here: ${clangTidyUrl}"
     fi
 
-    if [ ! -f bin/clang-format -o ! -f bin/clang-tidy ]
-    then
+    if [ ! -f bin/clang-format -o ! -f bin/clang-tidy ]; then
         echo "Either Clang-tidy or clang-format was not installed. Please install and put them on the PATH to use jit-format."
         echo "Tools can be found at http://llvm.org/releases/download.html#3.8.0"
     fi
 }
 
-if ! dotnet --info; then 
-    echo "Can't find dotnet! Please add to PATH."
-    return 1
+# Start the non-functions.
+
+get_host_os
+
+# Check if our required tools exist.
+
+if ! hash dotnet 2>/dev/null; then
+    echo "Can't find dotnet! Please install from http://dot.net and add to PATH."
+    exit 1
 fi
 
-if ! git --version; then 
+if ! hash git 2>/dev/null; then
     echo "Can't find git! Please add to PATH."
-    return 1
+    exit 1
 fi
 
-root=$(pwd)
-
-# Clone the mcgutils repo
-
-git clone https://github.com/dotnet/jitutils.git
+# Are we already in the dotnet/jitutils repo? Or do we need to clone it? We look for build.cmd
+# in the directory this script was invoked from.
+
+# Obtain the location of the bash script.
+__root="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+# Check if the bootstrap script is in the root of the jitutils repo.
+# By default, we're going to clone the repo.
+__clone_repo=1
+if [ -e ${__root}/build.cmd ]; then
+    # We found build.cmd. But make sure it's the root of the repo.
+    __root="$( cd ${__root} && git rev-parse --show-toplevel )"
+    pushd ${__root} >/dev/null
+    git remote -v | grep "/jitutils" >/dev/null
+    if [ $? == 0 ]; then
+        # We've proven that we're at the root of the jitutils repo.
+        __clone_repo=0
+    fi
+    popd >/dev/null
+fi
 
-pushd ./jitutils
+if [ ${__clone_repo} == 1 ]; then
+    git clone https://github.com/dotnet/jitutils.git
+    pushd ${__root}/jitutils >/dev/null
+else
+    pushd . >/dev/null
+fi
 
-# Pull in needed packages.  This works globally. (due to global.json)
+# Pull in needed packages.  This works globally (due to global.json).
```

Comment:
Looks like I missed this comment when migrating to csproj. There isn't a global.json  anymore, it's due to jitutils.sln now.